src/Pages/HomePage/HomePage.js

```diff
@@ -1,21 +1,79 @@
-import React  from 'react';
+import React , { useState }  from 'react';
 import { SideBar, CategoryBar } from '../../components/index';
-import { useVideos } from '../../context/index';
+import { useVideos , useHistoryContext , useWatchlaterContext , usePlayListContext, useToastContext, useUserAuth } from '../../context/index';
 import './HomePage.css';
 import { Link } from 'react-router-dom';
-import { addVideoInHistory } from '../../api-calls/api-calls';
-import { useHistoryContext } from '../../context/HistoryContext';
+import { addVideoInHistory , addToWatchLater, removeFromWatchLater , createPlayList , addVideoToPlaylist , removeVideoFromPlaylist } from '../../api-calls/api-calls';
 
 export function HomePage() {
   const { historyVideosDispatch } = useHistoryContext();
+  const { watchLaterDispatch } = useWatchlaterContext();
+  const { filteredVideos , setFilteredVideos , setVideos } = useVideos();
+  const [showModal , setShowModal ] = useState(false);
+  const [playlistName , setPlaylistName ] = useState('');
+  const { playListState , playListDispatch } = usePlayListContext();
+  const { isLogIn } = useUserAuth();
+  const notify = useToastContext();
+  let playlists;
+    if(playListState.payload === undefined || playListState.payload === 'none'){
+      playlists = [];
+    }else{
+      playlists = playListState.payload.playlist.data.playlists;
+    }
+
+  const updatePlayList = async(e, playlistId , videoData) => {
+    if(e.target.checked){
+       await addVideoToPlaylist(videoData , playlistId); 
+    }else{
+      await removeVideoFromPlaylist(playlistId , videoData._id);       
+    }
+  }
+
+  const createPlaylist = async() => {
+    if(playlistName){
+      const response = await createPlayList(playlistName);
+      playListDispatch({type : 'NEW_PLAYLIST', payload: response})
+    }else{
+      notify('Please Enter Playlist Name!',{type:'warning'})
+    }
+    setPlaylistName('');
+  }
+
+  const closeModal = () => {
```

Comment:
can use anonymous function right adjacent to event listener as you're using this only once in your file